xarray/tests/test_datatree.py

```diff
@@ -729,4 +728,99 @@ def test_filter(self):
             name="Abe",
         )
         elders = simpsons.filter(lambda node: node["age"].item() > 18)
-        dtt.assert_identical(elders, expected)
+        assert_identical(elders, expected)
+
+
+class TestDSMethodInheritance:
```

Comment:
This felt like a good place for adding these tests. But I could move them to another module if that's a preference (just probably not under the name `test_dataset_api.py`, which might be confusing outside of a specifically `datatree` oriented context).